packages/react-components/react-progress/src/components/Progress/useProgressState.tsx

```diff
@@ -0,0 +1,35 @@
+import type { ProgressState, ProgressProps } from './Progress.types';
+import { progressCssVars } from './useProgressStyles';
+
+// if the percentComplete is near 0, don't animate it.
+// This prevents animations on reset to 0 scenarios
+const ZERO_THRESHOLD = 0.01;
+
+export const useProgressState_unstable = (state: ProgressState, props: ProgressProps) => {
+  const { percentComplete = -1 } = props;
+
+  const determinate = percentComplete > -1 ? true : false;
+
+  const valuePercent = determinate ? Math.min(100, Math.max(0, percentComplete)) : undefined;
+
+  const progressBarStyles = {
+    [`${progressCssVars.percentageCssVar}`]: determinate ? valuePercent + '%' : '0',
+    [`${progressCssVars.transitionCssVar}`]:
+      valuePercent && determinate && valuePercent < ZERO_THRESHOLD ? 'none' : 'width .3s ease',
```

Comment:
It might be better to have 2 class names rather than a custom transition value in code.

Reply:
I agree, this is logic that should go in `useProgressStyles` even if it leads to more entries in the `makeStyles` call.

Reply:
I've moved the logic to `useProgress` for now, but the value of the width of the determinate `Progress` is gotten from `percentComplete`, which is a prop that has to be passed into the `useProgressStyles` to set the value